### ISSUE TITLE
[Feature Request] If chasing a VIPokemon or priority > XYZ, ignore unnecessary tasks to save time

### ISSUE THREAD

**Issue by Raiven66** [PokemonGoF/PokemonGo-Bot#2481]
ignore tasks like
- catching other pokemon
- spinning
- ...

**Comment by alexyaoyang** [comment]
Why not put CatchVisiblePokemon and CatchLuredPokemon at the top of your task list so the bot prioritize that first?

**Comment by Raiven66** [comment]
the bot runs ALL the tasks every step right? so how does putting them higher in the list help?

**Comment by alexyaoyang** [comment]
The bot executes tasks from the top to the bottom. If you put catching at the top, the bot will catch first instead of spin or recycle items.

**Comment by Raiven66** [comment]
but that has nothing to do with the request:
I proposed a feature, where if the bot is going after pokemon via PokemonGO-Map, it sometimes wastes time on the road because of random pokemon encounters, spinning forts and so on. It happened to me several times that because of these timewastes, the pokemon wasn't in reach anymore before it disappears.
For that matter it would be great if while going after important pokemon, the bot would not spin, catch random pokemon, etc.

**Comment by alexyaoyang** [comment]
What you are asking for are:
1) Going after important pokemon, ignoring lower priority ones (#1992 settles this once it is merged)
2) Not spinning forts first (place CatchVisiblePokemon and CatchLuredPokemon at the top of your task lists settles this)

**Comment by Raiven66** [comment]
I am on the dev branch, so #1992 i already am using
ad 1) going after Pokemon according to their priority in config is working fine - but while going after VIPs, ignoring other tasks would make sense for a higher chance of catching them before the time limit runs out
ad 2) since all tasks are being run every step, moving them in order doesn't make a difference - the proposal is to not spin and ignore catching other pokemon on the way other than the particular important pokemon from point 1), that means only ignore them under certain condition

**Comment by alexyaoyang** [comment]
A] skip spinning and catching pokes if going after an important pokemon via PokeGo-Maps
Fix it by setting snipe to true on #1992.

**Comment by Raiven66** [comment]
Setting snipe to true transports me there directly, without imitating human behaviour (walking) - easily discoverable and bannable, since snipe feature as of #1992 isn't configurable to do that.
That i proposed an improved sniping feature in #2480.

**Comment by alexyaoyang** [comment]
So you want to **walk** to important pokemon and ignore everything on the way? 

You'll be limited to very a small area to snipe, since normal walking speed is 4km/h, and jogging is 12km/h, which in turn decreases your chances of encountering a rare pokemon quite drastically. 

If you increase your walking speed too much, then it is not imitating human behaviour as well.

Just my 2 cents. Maybe someone else will have another opinion :)

**Comment by Raiven66** [comment]
actually, the speed of movement (at least in imaginable margins) is I think irrelevant to catching pokemon, it just doesn't let you hatch eggs. I know several people driving a car full of people to catch pokemon around the city, which is probably not bannable. Teleporting to snipe longer distances is on the other hand suspicious behavior. If you would not catch the pokemon and only teleport, it could be said to be a GPS malfunction and thus overlooked, but the action of encountering and probably catching the things seems too easy to discover to me.

For the whole yesterday the bot has been running for me in a step 6 area around my city, collecting pokemon which appeared with the help of PoGO-map, creating a very random pattern on the map that the FollowSpiral or FollowPath are unable to provide due to the repetitive nature of the alhorithm. This random behavior was in my opinion very difficult to discover.
With this in mind, since the bot was following some pokemon for quite a long time while ignoring others (priority -> expected behav.), it did stumble upon some forts and random pokemon encounters which then rendered the whole trip (±10mins) to the pokemon useless, since time ran out before the bot could get there (at walking speed config of 6).

So the upside of this feature would be

-  randomly looking walking path pattern
-  catching rarer pokemons with higher probability while maintaining humanlike behavior
-  time saving
